Crypto/Cipher/ChaChaPoly1305.hs

```diff
@@ -81,6 +81,18 @@ instance ByteArrayAccess Nonce where
   withByteArray (Nonce8  n) = B.withByteArray n
   withByteArray (Nonce12 n) = B.withByteArray n
 
+instance Eq Nonce where
+  (Nonce8  a) == (Nonce8  b) = a == b
+  (Nonce12 a) == (Nonce12 b) = a == b
+  (Nonce8  a) == (Nonce12 b) = a == b
+  (Nonce12 a) == (Nonce8  b) = a == b
+
+instance Ord Nonce where
+  compare (Nonce8  a) (Nonce8  b) = compare a b
+  compare (Nonce12 a) (Nonce12 b) = compare a b
+  compare (Nonce8  a) (Nonce12 b) = compare a b
```

Comment:
when they are considered as integer yes, but here I'm saying to basically consider them as 2 independent spaces. It's probably a good idea to be consistent with == where
`Nonce8 0000 0000` would be not equal to `Nonce12 0000 0000 0000`. if you compare the byte in `Ord`, then you run into the problem of having something that is not `==` but `compare == EQ` hold